From 7eaf65d5e1c1872afb006a0f706ab24fe12049b2 Mon Sep 17 00:00:00 2001
From: Thomas Orozco <thomas@orozco.fr>
Date: Tue, 27 Feb 2018 12:25:59 +0100
Subject: [PATCH 2/2] Elasticsearch: Set Content-Type

Content-Type is required starting with Elasticsearch 6. The better
solution here would be for us to update to a newer Logstash... but on
the other hand, we want to move off of Logstash altogether, so this is
not worth doing right now.
---
 lib/logstash/outputs/elasticsearch/protocol.rb | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/logstash/outputs/elasticsearch/protocol.rb b/lib/logstash/outputs/elasticsearch/protocol.rb
index 3c982a3..d7a1e73 100644
--- a/lib/logstash/outputs/elasticsearch/protocol.rb
+++ b/lib/logstash/outputs/elasticsearch/protocol.rb
@@ -81,12 +81,13 @@ module LogStash::Outputs::Elasticsearch
             :request_timeout => 0  # and requests
           },
           :transport_class => transport_class,
-          :pipeline => options[:pipeline]
+          :pipeline => options[:pipeline],
+          :headers => { 'Content-Type' => 'application/json' }
         }
 
         if options[:user] && options[:password] then
           token = Base64.strict_encode64(options[:user] + ":" + options[:password])
-          client_options[:headers] = { "Authorization" => "Basic #{token}" }
+          client_options[:headers]["Authorization"] = "Basic #{token}"
         end
 
         Elasticsearch::Client.new client_options
-- 
2.14.1

